src/postgres/inventory/type/PgClassType.ts

```diff
@@ -80,9 +80,21 @@ class PgClassType extends PgType<PgRow> implements ObjectType<PgRow> {
     if (typeof pgValue !== 'object')
       throw new Error(`Postgres value of object type must be an object, not '${typeof pgValue}'.`)
 
-    return new Map(Array.from(this.fields).map<[string, mixed]>(([fieldName, field]) => {
-      return [fieldName, field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])]
-    }))
+    // Parse official pg fields
+    const obj = {};
+    Array.from(this.fields).filter<[string, mixed]>(([fieldName, field]) => pgValue.hasOwnProperty(field.pgAttribute.name)).map<[string, mixed]>(([fieldName, field]) => {
+      obj[fieldName] = field.type.transformPgValueIntoValue(pgValue[field.pgAttribute.name])
+    })
+    // Pass through other fields unchanged
+    for (const key in pgValue) {
+      obj[key] = pgValue[key];
+    }
+
+    const pairs = [];
+    for (const key in obj) {
```

Comment:
Thanks, I've not played with maps yet so probably ought to spend some time reading the docs.

Incidentally why did you use maps here rather than just vanilla JS objects?